lib/Screens/ChatPage/ChatPage.dart

```diff
@@ -1,190 +1,75 @@
-      source: ImageSource.gallery,
-    );
-
-    if (result != null) {
-      _setAttachmentUploading(true);
-      final file = File(result.path);
-      final size = file.lengthSync();
-      final bytes = await result.readAsBytes();
-      final image = await decodeImageFromList(bytes);
-      final name = result.name;
-
-      try {
-        final reference = FirebaseStorage.instance.ref(name);
-        await reference.putFile(file);
-        final uri = await reference.getDownloadURL();
-
-        final message = types.PartialImage(
-          height: image.height.toDouble(),
-          name: name,
-          size: size,
-          uri: uri,
-          width: image.width.toDouble(),
-        );
+  @override
+  void initState() {
+    getUserEmail();
 
-        FirebaseChatCore.instance.sendMessage(
-          message,
-          widget.room.id,
-        );
-        _setAttachmentUploading(false);
-      } finally {
-        _setAttachmentUploading(false);
-      }
-    }
+    super.initState();
   }
 
-  void _handleMessageTap(BuildContext context, types.Message message) async {
-    if (message is types.FileMessage) {
-      var localPath = message.uri;
-
-      if (message.uri.startsWith('http')) {
-        final client = http.Client();
-        final request = await client.get(Uri.parse(message.uri));
-        final bytes = request.bodyBytes;
-        final documentsDir = (await getApplicationDocumentsDirectory()).path;
-        localPath = '$documentsDir/${message.name}';
-
-        if (!File(localPath).existsSync()) {
-          final file = File(localPath);
-          await file.writeAsBytes(bytes);
-        }
-      }
-
-      await OpenFile.open(localPath);
-    }
+  getUserEmail() async {
+    await HelperFunctions.getUserEmailInSharedPreference().then((email) {
+      setState(() {
+        this._email = email;
+        this.getChatHistory();
+      });
+    });
   }
 
-  void _handlePreviewDataFetched(
-    types.TextMessage message,
-    types.PreviewData previewData,
-  ) {
-    final updatedMessage = message.copyWith(previewData: previewData);
-
-    FirebaseChatCore.instance.updateMessage(updatedMessage, widget.room.id);
+  void getChatHistory() {
+    ChatService()
+        .getChatHistory(this._email, widget.channel.idChannel)
+        .then((value) {
+      setState(() {
+        this._messages = value;
+      });
+    });
   }
 
   void _handleSendPressed(types.PartialText message) {
-    FirebaseChatCore.instance.sendMessage(
-      message,
-      widget.room.id,
+    final textMessage = types.TextMessage(
+      author: types.User(
+        id: widget.channel.idUser.toString(),
+      ),
+      id: const Uuid().v4(),
+      text: message.text,
     );
-  }
 
-  void _setAttachmentUploading(bool uploading) {
-    setState(() {
-      _isAttachmentUploading = uploading;
+    ChatService()
+        .sendMessage(message.text, widget.channel.idUser.toString(),
+            widget.channel.idChannel)
+        .then((value) {
```

Comment:
Quando vc não for usar o valor gerado vc pode usar (_) ao inves de (value) para indicar que o valor de retorno não tem importancia